azure_functions_worker/bindings/meta.py

```diff
@@ -25,15 +29,33 @@ def load_binding_registry() -> None:
     global BINDING_REGISTRY
     BINDING_REGISTRY = func.get_binding_registry()
 
-
-def get_binding(bind_name: str) -> object:
-    binding = None
-    registry = BINDING_REGISTRY
-    if registry is not None:
-        binding = registry.get(bind_name)
+    if BINDING_REGISTRY is None:
+        # If the BINDING_REGISTRY is None, azure-functions hasn't been
+        # loaded in properly.
+        raise AttributeError("BINDING_REGISTRY is None.")
+
+    # The base extension supports python 3.8+
+    if sys.version_info.minor >= BASE_EXT_SUPPORTED_PY_MINOR_VERSION:
+        # Import the base extension
+        try:
+            import azure.functions.extension.base as clients
+            global DEFERRED_BINDINGS_REGISTRY
+            DEFERRED_BINDINGS_REGISTRY = clients.get_binding_registry()
+        except ImportError:
+            # This means that the customer hasn't imported the library.
+            # This isn't an error.
+            pass
```

Comment:
Okay

Do we keep account of which app is using this feature and which isnt?

Reply:
![image](https://github.com/Azure/azure-functions-python-worker/assets/12449837/da6c3f3a-ce07-472b-b153-a9103b46dcfc)
worker installs base package for 3.8+ in setup.py regardless the feature is used or not. it shall not raise import error here otherwise a serious issue with worker build, customers whether using feature or not shall expect functions not running. so i think we should fail fast here @hallvictoria  @vrdmr